modules/openapi-generator/src/main/resources/go-server/model.mustache

```diff
@@ -17,7 +17,39 @@ const (
 	{{#enumClassPrefix}}{{{classname.toUpperCase}}}_{{/enumClassPrefix}}{{name}} {{{classname}}} = {{{value}}}
 	{{/enumVars}}
 	{{/allowableValues}}
-){{/isEnum}}{{^isEnum}}{{#description}}
+)
+
+// All allowed values of {{{classname}}} enum
+var Allowed{{{classname}}}EnumValues = []{{{classname}}}{
```

Comment:
Could we do a 

map[{{classname}}]struct{}

That should work in O(1) time

Reply:
Let me know if you need clarifications about this

Reply:
I changed it to be a map. I had to add in a call to the golang maps package for the error message when an invalid Enum was provided.